src/main/java/org/openstreetmap/atlas/geography/atlas/items/Containable.java

```diff
@@ -0,0 +1,21 @@
+package org.openstreetmap.atlas.geography.atlas.items;
+
+import org.openstreetmap.atlas.geography.GeometricSurface;
+
+/**
+ * Represents entities that can be contained inside a {@link GeometricSurface}.
+ *
+ * @author Yazad Khambata
+ */
+public interface Containable
```

Comment:
this is just my personal design opinion but all located features should be containable. So I don't see enough value to define an interface for it.

Reply:
The original plan was to implement this change in Located interface. However, the implications of doing so was wide reaching with about ~50 classes directly/indirectly needed an implementation for within. In some cases the implementation was not obvious, so discussed with @lucaspcram and added a new interface, which was a safe option that met the goal.

Reply:
I agree that this is not an ideal decision from a design standpoint - but in the interest of getting this functionality off the ground it makes some sense. I think it may be worth creating a TODO item to eventually change this to `Located`.

Reply:
@lucaspcram and @yiqingj - how about we add `within` to `Located` with a default implementation that checks based on `Located#bounds()`. If a `Located` implementation wants to improve the within check it was be overwrite this change there.

Reply:
Oh that's actually a great idea. Then you can only implement the more granular functionality where you need it. Just make sure this is clear in the doc comments, so future users can implement a more granular check should they need it.

Reply:
@lucaspcram @yazad3 the bounds implementation could be a temporary solution. Just from an end user point of view, if I were a user to use this method, I'd rather see it unimplemented than having unexpected behavior.  `Within` to me has a very clear definition and using bound will cause false negative result, imagining a user not paying close attention to the doc comments and has to debug why the result is not correct and eventually found out it's caused by this, I would definitely be angry if I were that person. :)
that being said having a `withinBound` method doesn't sound like a bad idea.

Reply:
@yiqingj and @matthieun - Thanks I am adding a default implementation that throws `UnsupportedOperationException`.